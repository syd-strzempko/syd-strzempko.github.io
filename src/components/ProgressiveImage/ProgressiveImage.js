import { useState, useEffect } from 'react';
import styles from './ProgressiveImage.module.css';

const ProgressiveImage = ({ image }) => {
  const [imgSrc, setImgSrc] = useState(image.compressed.pathLong || image.pathLong);

  useEffect(() => {
    const img = new Image();
    img.src = image.pathLong;
    img.onload = () => {
      setImgSrc(image.pathLong);
    };
  }, [image.pathLong]);

  return (
    <img src={imgSrc} className={styles.image} alt={`artwork ${image.id}`}/>
  );
}

export default ProgressiveImage;