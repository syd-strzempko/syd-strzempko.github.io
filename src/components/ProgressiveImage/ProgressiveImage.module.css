.tile {
    width: 100%;
    height: 100%;
    object-fit: cover;
}
.image {
    max-width: 100%;
    max-height: 100%;
    object-fit: contain;
}
.carousel {
    object-fit: contain;
    object-position: bottom;
    max-height: 90vh;
}
@media screen and (max-width: 600px) {
    .carousel {
        height: 100%;
        object-fit: cover;
        object-position: center;
    }
}