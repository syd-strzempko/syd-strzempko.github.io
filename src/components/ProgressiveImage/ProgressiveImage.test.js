import ProgressiveImage from '../ProgressiveImage';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

// Note: MemoryRouter allows for nested Link components to exist inside of & access stack of mocked router

const MOCKIMAGE = {
  pathLong: 'mock_long.jpg',
  pathShort: 'mock_short.jpg',
  id: 0,
  compressed: {
    pathLong: 'mock_long.jpg',
    pathShort: 'mock_short.jpg'
  }
}

describe('ProgressiveImage', () => {
  it('renders correctly', () => {
    let view = renderer.create(
      <MemoryRouter>
        <ProgressiveImage image={MOCKIMAGE} />
      </MemoryRouter>
    );
    expect(view.toJSON()).toMatchSnapshot();
  });
});