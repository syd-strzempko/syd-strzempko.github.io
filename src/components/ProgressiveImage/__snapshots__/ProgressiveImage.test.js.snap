// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressiveImage renders correctly 1`] = `
<img
  alt="artwork 0"
  className="image"
  src="mock_long.jpg"
/>
`;
